java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -462,7 +463,20 @@ private void executeMethodInvocation(MethodInvocationTree mit) {
     ProgramState.Pop unstack = programState.unstackValue(mit.arguments().size() + 1);
     programState = unstack.state;
     logState(mit);
-    programState = programState.stackValue(constraintManager.createMethodSymbolicValue(mit, unstack.values));
+    final SymbolicValue resultValue = constraintManager.createMethodSymbolicValue(mit, unstack.values);
+    programState = programState.stackValue(resultValue);
+    if (isNonNullMethod(mit)) {
+      programState = programState.addConstraint(resultValue, ObjectConstraint.NOT_NULL);
+    }
+  }
+
+  private static boolean isNonNullMethod(MethodInvocationTree mit) {
+    Symbol methodSymbol = mit.symbol();
+    if (methodSymbol.isUnknown()) {
+      return false;
+    }
+    JavaSymbol symbol = (JavaSymbol) methodSymbol;
+    return symbol.metadata().isAnnotatedWith("javax.annotation.Nonnull");
```

Comment:
No need to cast to `JavaSymbol` to access the metadada.  In fact the method can even be refactored as : 
```
private static boolean isNonNullMethod(Symbol symbol) {
  return !symbol.isUnknown() && symbol.metadata().isAnnotatedWith("javax.annotation.Nonnull");
}
```